### PR TITLE
fix(claude): auto-approve bypass tool permissions

### LINKED ISSUE #263: Edit/Write tool calls blocked despite permission_mode=bypassPermissions in IM platform flow
## Summary

When using vibe-remote to interact with Claude Agent SDK via Discord (likely other IM platforms too), `Edit` and `Write` tool calls are blocked with a permission prompt that the user cannot answer — since IM platforms don't render the Claude Code permission UI — despite `permission_mode=bypassPermissions` being correctly set in the SDK invocation. `Bash` tool calls are NOT blocked, only file-write tools, indicating the gate is specific to Edit/Write.

## Environment

- vibe-remote: latest (uv tool install)
- Python: 3.13
- Platform: Discord
- OS: macOS 25.4.0
- Claude Agent SDK: bundled

## Reproduction

1. Standard vibe-remote Discord setup
2. Ask Claude to edit a file in a session
3. `Edit(...)` fails with: `Claude requested permissions to write to ..., but you haven't granted it yet.`
4. No UI in Discord to grant — request hangs
5. Same session: `Bash(cat > file)` works fine

## Verified state (rules out config issues)

- `ps` confirms SDK process launched with `--permission-mode bypassPermissions`
- Service log: `permission_mode: bypassPermissions`
- `~/.vibe_remote/state/settings.json` has correct `custom_cwd` for the channel
- Project `.claude/settings.json` has matching `Edit(.claude/agents/**/*.md)` allow rules
- `setting_sources` is `[user, project, local]` — settings.json should be honored
- Adding more rules to `.claude/settings.local.json` does not help

## Suspected cause

Either:
- The bundled `claude` binary has an internal safety layer that gates Edit/Write regardless of `--permission-mode`, OR
- vibe-remote needs an explicit `can_use_tool` callback for headless / IM-platform flows

## Workaround

Force file-modifying intents through Bash + python3 heredoc:

```python
python3 <<'PY'
content = open(path).read()
open(path, 'w').write(content.replace(OLD, NEW))
PY
```

This works fully — it's only the SDK's structured `Edit`/`Write` tools that are gated.

## Suggested fixes (any of)

1. **Add `can_use_tool` callback** in `core/handlers/session_handler.py:583` so headless / IM flows auto-approve:
   ```python
   from claude_agent_sdk import PermissionResultAllow
   async def _headless_approve(tool_name, tool_input, context):
       return PermissionResultAllow()
   option_kwargs["can_use_tool"] = _headless_approve
   ```

2. **Add `extra_args=["--dangerously-skip-permissions"]`** if `disableBypassPermissionsMode` is not set globally.

3. **Expose permission strategy in `ClaudeConfig`**:
   ```python
   @dataclass
   class ClaudeConfig:
       ...
       permission_strategy: str = "auto_approve"  # auto_approve | prompt_via_dm | strict
   ```

## Code references

- `config/v2_compat.py:82` — `permission_mode="bypassPermissions"` hardcoded but apparently insufficient
- `core/handlers/session_handler.py:561-583` — `ClaudeAgentOptions` construction site
- `core/handlers/session_handler.py:566` — `setting_sources: ["user", "project", "local"]`

Happy to test any patch.

### PR DESCRIPTION
**Pull request by cyhhao**
## What

Fixes #263.

Vibe Remote now treats Claude Code remote sessions as a fully headless bypass runtime:
- Always passes `permission_mode="bypassPermissions"` when creating Claude SDK sessions.
- Always attaches a `can_use_tool` callback that returns `PermissionResultAllow()` for every Claude tool permission request.
- Keeps remote-only interactive tools disabled (`AskUserQuestion`, `EnterPlanMode`, `ExitPlanMode`) because IM sessions cannot service those UI flows.

## Why

`bypassPermissions` covers ordinary file edits, but Claude Code can still ask for explicit permission for protected paths such as `.claude/agents/*.md` and `~/.claude/CLAUDE.md`. In an IM/headless runtime there is no permission UI to answer, so the session hangs.

The intended Vibe Remote product behavior is simpler: if the user is running Claude through Vibe Remote, the session is already a remote automation context and should bypass all Claude tool permission prompts. Vibe Remote should not try to replicate Claude Code's managed-settings policy parser or gate this behind an additional Vibe-specific env var.

## Validation

Evidence layers:
- Unit: `tests/test_claude_cli_path.py` verifies Claude sessions are forced to `bypassPermissions`, that the permission callback is attached, and that it allows generic tool permission requests. `tests/test_claude_sdk_compat.py` verifies the no-SDK `PermissionResultAllow` fallback is non-throwing.
- Contract: focused session option construction and SDK compatibility tests cover the SDK option/result surface.
- Scenario: no catalog scenario added; this is a Claude SDK permission callback edge case without an existing scenario ID.
- Manual: prior SDK probes confirmed `can_use_tool -> PermissionResultAllow()` unblocks `.claude/agents/*.md`, while ordinary files already work under `bypassPermissions`.

Checks run:
- `uv run --no-sync pytest tests/test_claude_cli_path.py tests/test_claude_sdk_compat.py -q`
- `uv run --no-sync ruff check core/handlers/session_handler.py modules/claude_sdk_compat.py tests/test_claude_cli_path.py tests/test_claude_sdk_compat.py`

## Risk

This deliberately bypasses Claude Code tool permission prompts for Vibe Remote Claude sessions. That matches the remote automation model, but deployments that need Claude Code's interactive or organization-managed permission gates should not route Claude through Vibe Remote with this backend behavior.

### CODE REVIEW

File: modules/claude_sdk_compat.py [hunk 1]
```diff
@@ -73,6 +74,9 @@ class ClaudeAgentOptions(_MissingClaudeSDK):
     class ClaudeSDKClient(_MissingClaudeSDK):
         pass
 
+    class PermissionResultAllow(_MissingClaudeSDK):
+        pass
```
[the remark this answers is not in the document]
Comment (cyhhao) [reply modules/claude_sdk_compat.py]: Fixed in 52693d1. `PermissionResultAllow` now has a non-throwing fallback with the same fields the callback needs, and `tests/test_claude_sdk_compat.py::test_missing_sdk_permission_allow_fallback_is_non_throwing` covers the missing-SDK import path.

**Comment by cyhhao** [comment]
@codex review

File: core/handlers/session_handler.py [hunk 2]
```diff
@@ -247,6 +250,40 @@ def _should_force_claude_sandbox(self) -> bool:
...
+        return permissions.get("disableBypassPermissionsMode") == "disable"
+
+    def _claude_bypass_disabled_by_settings(self, working_path: str) -> bool:
+        settings_paths = [
+            Path("/Library/Application Support/ClaudeCode/managed-settings.json"),
+            Path("/etc/claude-code/managed-settings.json"),
+            Path.home() / ".claude" / "settings.json",
+            Path(working_path) / ".claude" / "settings.json",
+            Path(working_path) / ".claude" / "settings.local.json",
+        ]
```
[the remark this answers is not in the document]
Comment (cyhhao) [reply core/handlers/session_handler.py]: Fixed in 2e602d1. The bypass-disable guard now checks the documented endpoint-managed sources before attaching `can_use_tool`: managed-settings drop-ins plus base files, macOS managed preferences, and Windows HKLM/HKCU policy JSON. Added focused tests for each managed source family.

**Comment by cyhhao** [comment]
@codex review

File: core/handlers/session_handler.py [hunk 3]
```diff
@@ -247,6 +253,122 @@ def _should_force_claude_sandbox(self) -> bool:
...
+        paths: list[Path] = []
+        for root in self._claude_managed_settings_roots():
+            dropin_dir = root / "managed-settings.d"
+            if dropin_dir.exists():
+                paths.extend(sorted(dropin_dir.glob("*.json")))
+            paths.append(root / "managed-settings.json")
+        paths.extend(
+            [
+                # Legacy Windows file-based managed settings path used before Claude Code v2.1.75.
+                Path(os.environ.get("ProgramData", r"C:\ProgramData")) / "ClaudeCode" / "managed-settings.json",
```
[the remark this answers is not in the document]
Comment (cyhhao) [reply core/handlers/session_handler.py]: Fixed in a5a5091. Removed the legacy `ProgramData\\ClaudeCode\\managed-settings.json` probe so stale pre-v2.1.75 files cannot suppress the callback.

File: core/handlers/session_handler.py [hunk 4]
```diff
@@ -247,6 +253,122 @@ def _should_force_claude_sandbox(self) -> bool:
...
+        if os.name != "nt":
+            return []
+        try:
+            import winreg  # type: ignore[import-not-found]
+        except Exception as exc:
+            logger.warning("Failed to import winreg for Claude managed settings inspection: %s", exc)
+            return []
+
+        settings: list[dict[str, Any]] = []
+        for hive in (winreg.HKEY_LOCAL_MACHINE, winreg.HKEY_CURRENT_USER):
```
[the remark this answers is not in the document]
Comment (cyhhao) [reply core/handlers/session_handler.py]: Fixed in a5a5091. Windows policy inspection now returns the first available registry policy source only, preserving HKLM precedence over HKCU instead of treating any lower-priority HKCU value as authoritative.

**Comment by cyhhao** [comment]
@codex review

File: core/handlers/session_handler.py [hunk 5]
```diff
@@ -576,6 +695,11 @@ def _capture_claude_stderr(line: str) -> None:
             "stderr": _capture_claude_stderr,
             "max_buffer_size": CLAUDE_SDK_MAX_BUFFER_SIZE,
         }
+        if (
+            self.config.claude.permission_mode == "bypassPermissions"
+            and not self._claude_bypass_disabled_by_settings(working_path)
+        ):
+            option_kwargs["can_use_tool"] = self._allow_claude_bypass_tool
```
[the remark this answers is not in the document]
Comment (cyhhao) [reply core/handlers/session_handler.py]: Fixed in 99780d8. Vibe Remote no longer attaches the auto-allow callback by default because Claude Team/Enterprise remote managed settings are not exposed through the Python SDK. The callback now requires an explicit local opt-in via `VIBE_REMOTE_CLAUDE_AUTO_ALLOW_BYPASS_TOOLS=1`, still respects locally visible disable-bypass settings before attaching, and denies unexpected permission requests instead of returning allow for every tool.

**Comment by cyhhao** [comment]
@codex review

File: core/handlers/session_handler.py [hunk 6]
```diff
@@ -247,6 +255,153 @@ def _should_force_claude_sandbox(self) -> bool:
...
+                return True
+        for settings in self._iter_claude_windows_registry_settings():
+            if self._settings_disable_claude_bypass(settings):
+                logger.info("Claude bypassPermissions mode is disabled by Windows policy settings")
+                return True
+
+        for settings_path in self._iter_claude_file_settings_paths(working_path):
+            if not settings_path.exists():
+                continue
+            try:
```
[the remark this answers is not in the document]
Comment (cyhhao) [reply core/handlers/session_handler.py]: Resolved in 28ba35b by removing the Claude settings-policy parser entirely. The product decision is that Vibe Remote Claude sessions are fully headless bypass sessions: we always pass `permission_mode="bypassPermissions"` and always return `PermissionResultAllow()` from `can_use_tool`, so there is no managed-source precedence scan to get wrong.

**Comment by cyhhao** [comment]
@codex review
